docs: clarify breaking changes to `GOSUMDB` in Renovate v38 (#34158)
renovatebot/renovate: docs/usage/release-notes-for-major-versions.md

- **git:** check _all_ commits on the branch to decide if the branch was modified ([#28225](https://github.com/renovatebot/renovate/pull/28225))
- **gitea:** use "bearer auth" instead of "token auth" to authenticate to the Gitea platform
- **github:** if you run Renovate as a GitHub app then `platformCommit` is automatically enabled

> - **gomod:** the value of `GOSUMDB` has changed, which will now check signatures for modules. If you are using Renovate with private Go modules, you will need to set `GOPRIVATE`. For more details, see [Go's official documentation for working with private Go modules](https://golang.org/ref/mod#private-modules).

Reviewer: ```suggestion
- **gomod:** the value of `GOSUMDB` is no longer set to `off`, therefore Go will now check signatures for modules. If you are using Renovate with private Go modules, you will need to set `GOPRIVATE`. For more details, see [Go's official documentation for working with private Go modules](https://golang.org/ref/mod#private-modules).
```
IMO more descriptive, of what is happening

Author: Maybe a little too much detail, but thoughts? 
```suggestion
- **gomod:** the value of `GOSUMDB` was previously set to `off`, meaning the Go toolchain would not validate signatures for modules. This has now been corrected, which may result in errors updating Go modules. In particular, if you are using Renovate with private Go modules, you will need to set `GOPRIVATE`. For more details, see [Go's official documentation for working with private Go modules](https://golang.org/ref/mod#private-modules).
```

Reviewer: SGTM
